src/FluentValidation.AspNetCore/RuleSetForClientSideMessagesPageFilter.cs

```diff
@@ -0,0 +1,47 @@
+#if NETCOREAPP3_1 || NET5_0
+
+#region License
+// Copyright (c) .NET Foundation and contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// The latest version of this file can be found at https://github.com/FluentValidation/FluentValidation
+#endregion
+
+namespace FluentValidation.AspNetCore {
+	using System.Linq;
+	using System.Threading.Tasks;
+	using Microsoft.AspNetCore.Mvc.Filters;
+
+	/// <summary>
+	/// Determines which ruleset should be used when deciding which validators should be used to generate client-side messages.
+	/// </summary>
+	internal class RuleSetForClientSideMessagesPageFilter : IAsyncPageFilter {

```

Comment:
Changed this to internal too

Reply:
My only comment about this is that it would prevent people from adding it themselves to a specific page. The point of the configuration option was to allow removing the filter from all requests and adding it to only specific pages. If we aren't going with the option in the end they the filter can be internal for sure.

Reply:
Yeah let's remove the configuration option and leave the classes internal. If someone asks for this functionality then we can reconsider it in the future, but I prefer not to make the public api surface larger than necessary unless there's a good reason (as this increases my support burden).

Reply:
I am happy with that :)